### PR TITLE
Support reading YAML files with classic Java objects

### PR DESCRIPTION
**Pull request by pavelzubarev**

### CODE REVIEW

**Comment by pavelzubarev** [comment]
Side question: What is our current practice for consuming such SNAPSHOT packages (except the local build)? Should we publish snapshots as well?

**Comment by pavelzubarev** [comment]
More I look on it, more I think it should be part of my pipeline and not SDK at this stage, actually

File: src/main/java/com/cognite/beam/io/transform/yaml/ReadYamlObjectMap.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+     */
+    public static <U> ReadYamlObjectMap<U> from(String fileName, Class<U> klass) {
+        return from(ValueProvider.StaticValueProvider.of(fileName), klass);
+    }
+
+    abstract ReadYamlObjectMap.Builder<T> toBuilder();
+
+    abstract ValueProvider<String> getFilePath();
+
+    abstract Class<T> getKlass();
```
Comment (kjetilh33) [inline src/main/java/com/cognite/beam/io/transform/yaml/ReadYamlObjectMap.java:86]: Are we able to find a better name? "getTargetClass()", "getOutputClass()" or similar?

**Review by kjetilh33 — CHANGES_REQUESTED** [review]

**Comment by kjetilh33** [comment]
> Side question: What is our current practice for consuming such SNAPSHOT packages (except the local build)? Should we publish snapshots as well?

Per now we haven't published the SNAPSHOT packages. I don't have any specific preferences to be honest.

**Pull request closed without merging**